### PR TITLE
feat: add quality tasks - first approximation.

### PR DESCRIPTION
**Pull request by johanseto**
## Description

The PR purpose is to integrate quality tasks into the repo.

## Checklist for Merge

- [ ] Add  repo files: license File, readme. changelog
- [x] Update makefile
- [ ] Add setup configuration (setup.py setup.cfg)
- [x] Add lint or quality action.
- [x] Clean python code
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        make requirements
+    - name: Run the QA suite
+      run: |
+        make quality
+    # - name: Unit tests
+    #   run: |
+    #     make test
```
Comment (felipemontoya) [inline .github/workflows/test.yml]: Mind the EOF with a new line

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        make requirements
+    - name: Run the QA suite
+      run: |
+        make quality
+    # - name: Unit tests
```
Comment (felipemontoya) [inline .github/workflows/test.yml]: Dont leave things commented out. If you want it in, leave it, otherwise remove it.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: Unit tests, QA
+
+on:
+  push:
+    branches: [ main, quality-tasks ]
```
Comment (felipemontoya) [inline .github/workflows/test.yml]: Why not all?

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,671 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```
Comment (felipemontoya) [inline LICENSE]: Please open a PR only for this

File: Makefile [hunk 5]
```diff
@@ -1,7 +1,57 @@
...
+	#python setup.py bdist_wheel
+	make selfcheck
+
+requirements: ## install development environment requirements
+	pip install -r requirements/pip.txt
+	pip install -r requirements/pip-tools.txt
+	pip-sync requirements/dev.txt
+	#pip install -e .
+
+test: ## run unitary tests and meassure coverage
```
Comment (felipemontoya) [inline Makefile]: we can leave the unit testing bits for a different PR

File: Makefile [hunk 6]
```diff
@@ -1,7 +1,57 @@
...
+	make piptools
+	$(PIP_COMPILE) -o requirements/base.txt requirements/base.in
+	$(PIP_COMPILE) -o requirements/dev.txt requirements/dev.in
+
+quality: ## check coding style with pycodestyle and pylint
+	pylint codejailservice *.py
+	pycodestyle codejailservice *.py
+	pydocstyle codejailservice *.py
+	isort --check-only --diff --recursive codejailservice *.py
+	#python setup.py bdist_wheel
```
Comment (felipemontoya) [inline Makefile]: we need to fix this and uncomment

Comment (felipemontoya) [reply Makefile]: But also maybe fixing the setup deserves its own PR

File: codejailservice/__init__.py [hunk 7]
```diff
@@ -1 +1,2 @@
+"""Helps you keep your cool when creating dozens of open edX and eduNEXT environments."""
```
Comment (felipemontoya) [inline codejailservice/__init__.py]: This should be a docstring for this module.
Somehting like:



```suggestion
"""Allows for the remote execution of unsafe code via a rest API that runs restricted by an apparmor profile."""
```

File: codejailservice/app.py [hunk 8]
```diff
@@ -56,22 +62,25 @@ def configure_codejail(app):
...
+
 configure_codejail(app)
 
-from codejail.safe_exec import SafeExecException, json_safe
-from codejail.safe_exec import not_safe_exec as codejail_not_safe_exec
-from codejail.safe_exec import safe_exec as codejail_safe_exec
 
 @app.route("/")
 def index():
+    """Index route logic."""
```
Comment (felipemontoya) [inline codejailservice/app.py]:
```suggestion
    """A welcome message to give an idea of what this service entails."""
```

Can we remove the reference to edx? edx->openedx.

"An openedx codejail service developed by edunext."

File: codejailservice/tutor.py [hunk 9]
```diff
@@ -0,0 +1 @@
+"""Helps you keep your cool when creating dozens of open edX and eduNEXT environments."""
```
Comment (felipemontoya) [inline codejailservice/tutor.py:1]: do we need this file?

File: pylintrc [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+deprecated-modules = regsub,TERMIOS,Bastion,rexec
+import-graph =
+ext-import-graph =
+int-import-graph =
+known-third-party=codejail
+
+[EXCEPTIONS]
+overgeneral-exceptions = Exception
+
+# 1a67033d4799199101eddf63b8ed0bef3e61bda7
```
Comment (felipemontoya) [inline pylintrc]: EOF

File: setup.cfg [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+[wheel]
+universal = 1
+
+[pycodestyle]
+ignore=E501, E722
+exclude=.git,.tvm
+
+[tool:pytest]
+norecursedirs = strains .tvm
```
Comment (felipemontoya) [inline setup.cfg]: EOF

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+line_length = 120
+multi_line_output = 3
+known_third_party = codejail
+
+[wheel]
+universal = 1
+
+[pycodestyle]
+ignore=E501, E722
+exclude=.git,.tvm
```
Comment (felipemontoya) [inline setup.cfg]: no need to ignore tvm

File: setup.cfg [hunk 13]
```diff
@@ -0,0 +1,16 @@
+[isort]
+include_trailing_comma = True
+indent = '    '
+line_length = 120
+multi_line_output = 3
+known_third_party = codejail
+
+[wheel]
```
Comment (felipemontoya) [inline setup.cfg]: This belongs together with the setup.py PR

**Review by felipemontoya — CHANGES_REQUESTED** [review]
Overall a good PR, but we need to split it.
Lets leave this just for QA and open new PRs for Unit testing and setup.py

File: codejailservice/tutor.py (hunk 9 above)
Comment (johanseto) [reply codejailservice/tutor.py:1]: In the process, updating the machine to run the codejail service and plugin to resolve it.

File: pylintrc (hunk 10 above)
Comment (johanv26) [reply pylintrc]: Solved in https://github.com/eduNEXT/codejailservice/pull/3/commits/d881138a9a318bd21bb362a0c0a57c15dee8fdfc.

File: .github/workflows/test.yml (hunk 1 above)
Comment (johanseto) [reply .github/workflows/test.yml]: Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: .github/workflows/test.yml (hunk 2 above)
Comment (johanseto) [reply .github/workflows/test.yml]: Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: .github/workflows/test.yml (hunk 3 above)
Comment (johanseto) [reply .github/workflows/test.yml]: Changed to  `'**'` for check all branches in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: LICENSE (hunk 4 above)
Comment (johanseto) [reply LICENSE]: I quit the License file in this commit: 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f.
Next: open new PR

File: Makefile (hunk 5 above)
Comment (johanseto) [reply Makefile]: Quit this section in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: Makefile (hunk 6 above)
Comment (johanseto) [reply Makefile]: First: Quit this part in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f.

File: pylintrc (hunk 10 above)
Comment (johanseto) [reply pylintrc]: Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: setup.cfg (hunk 11 above)
Comment (johanseto) [reply setup.cfg]: Solved in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: setup.cfg (hunk 12 above)
Comment (johanseto) [reply setup.cfg]: Not ignored .tvm in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f

File: setup.cfg (hunk 13 above)
Comment (johanseto) [reply setup.cfg]: I quit the wheel section in 8c0b8b8ada32e5e7b0e15fc1b05f33deea2c503f.
Next: Make the setup PR

**Comment by johanseto** [comment]
This PR is moved or continue in PR: https://github.com/eduNEXT/codejailservice/pull/6
